include/abstract_graph_store.h

```diff
@@ -14,18 +13,55 @@ namespace diskann
 class AbstractGraphStore
 {
   public:
-    AbstractGraphStore(const size_t max_pts) : _capacity(max_pts)
+    AbstractGraphStore(const size_t total_pts, const size_t reserve_graph_degree)
+        : _capacity(total_pts), _reserve_graph_degree(reserve_graph_degree)
+    {
+    }
+
+    // returns tuple of <nodes_read, start, num_frozen_points>
+    virtual std::tuple<uint32_t, uint32_t, size_t> load(const std::string &index_path_prefix,
```

Comment:
does load need to know num_points?

Reply:
LoadGraph Method in Index class takes expected num points and params, and if that is larger then num points in graph we kind of resize the graph.